### PR TITLE
Update golang for docs/Dockerfile to 1.21

### PR DESCRIPTION
**Pull request by spolti**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

File: docs/apis/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,10 @@
-FROM golang:1.17
+FROM golang:1.21
 
 RUN apt-get update && apt-get -y upgrade && apt-get -y install git
 
 RUN git clone https://github.com/ahmetb/gen-crd-api-reference-docs
+# replace the version pattern in the Dockerfile until we don't move to golang 1.22
+RUN sed -i 's/1.21.0/1.21/g' gen-crd-api-reference-docs/go.mod
```
Comment (sivanantha321) [inline docs/apis/Dockerfile:7]: Why do we need this ?

Comment (spolti) [reply docs/apis/Dockerfile:7]: Because the Golang 1.22 pattern in the go.mod.
Golang 1.21 does not accept 1.21.0.

**Comment by spolti** [comment]
Hi, @yuzisun @sivanantha321 can you please take a look?

**Review by yuzisun — APPROVED** [review]

**Pull request merged**